user-operator/src/main/java/io/strimzi/operator/user/HealthCheckAndMetricsServer.java

```diff
@@ -0,0 +1,161 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.operator.user;
+
+import io.micrometer.prometheus.PrometheusMeterRegistry;
+import io.strimzi.operator.common.MetricsProvider;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.eclipse.jetty.server.Request;
+import org.eclipse.jetty.server.Server;
+import org.eclipse.jetty.server.handler.AbstractHandler;
+import org.eclipse.jetty.server.handler.ContextHandler;
+import org.eclipse.jetty.server.handler.ContextHandlerCollection;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+
+/**
+ * Jetty based web server used for health checks and metrics
+ */
+public class HealthCheckAndMetricsServer {
```

Comment:
There is currently no other use for this, so why should this be in the Cluster Operator's code base? For the other pieces I add to `operator-common` here, there is fairly clear use here (StrimziPodSet controller => I actually had it used there originally, but it is not inluded in this PR to not make it even bigger and to avoid conflicts). But the webserver has no value there at least until Vert.x is removed from there.

So yes, I can see this as a generic class in the future. But I think it is too far away to make it generic part of the `operator-common` already today.